chromium_src/content/browser/tld_ephemeral_lifetime.cc

```diff
@@ -41,27 +43,48 @@ TLDEphemeralLifetime::~TLDEphemeralLifetime() {
   storage_partition_->GetCookieManagerForBrowserProcess()->DeleteCookies(
       std::move(filter), base::NullCallback());
 
+  if (!on_destroy_callbacks_.empty()) {
+    auto on_destroy_callbacks = std::move(on_destroy_callbacks_);
+    for (auto& callback : on_destroy_callbacks) {
+      std::move(callback).Run(key_.second);
+    }
+  }
+
   active_tld_storage_areas().erase(key_);
 }
 
+// static
 TLDEphemeralLifetime* TLDEphemeralLifetime::Get(BrowserContext* browser_context,
                                                 std::string storage_domain) {
-  TLDEphemeralLifetimeKey key = std::make_pair(browser_context, storage_domain);
-  auto it = active_tld_storage_areas().find(key);
-  DCHECK(it == active_tld_storage_areas().end() || it->second.get());
-  return it != active_tld_storage_areas().end() ? it->second.get() : nullptr;
+  const TLDEphemeralLifetimeKey key(browser_context, std::move(storage_domain));
```

Comment:
same, `const&` instead of std::move and you still only have one copy operation inside `TLDEphemeralLifetimeKey`